src/View/Form/FormContext.php

```diff
@@ -94,7 +94,28 @@ public function val($field, $options = [])
             return $val;
         }
 
-        return $options['default'];
+        if ($options['default'] !== null || !$options['schemaDefault']) {
+            return $options['default'];
+        }
+
+        return $this->_schemaDefault($field);
+    }
+
+    /**
+     * Get default value from form schema for given field.
+     *
+     * @param string $field Field name.
+
+     * @return mixed
+     */
+    protected function _schemaDefault($field)
+    {
+        $field = $this->_form->schema()->field($field);
+        if ($field) {
+            return $field['default'];
```

Comment:
Should we isset() check on `$field['default']` instead of inferring the key based on the field existing.

Reply:
`default` key should always exist:

https://github.com/cakephp/cakephp/blob/ba8dfb1a9b284c335970a41ff9ff254e3189717c/src/Form/Schema.php#L71

Reply:
:+1: